VDA-StorageNode-Registry/contracts/StorageNodeRegistry.sol

```diff
@@ -0,0 +1,1098 @@
+    function depositToken(address didAddress, uint tokenAmount) external {
+        uint nodeId = _didNodeId[didAddress];
+        if (nodeId == 0) {
+            revert InvalidDIDAddress();
+        }
+
+        IERC20Upgradeable(vdaTokenAddress).transferFrom(tx.origin, address(this), tokenAmount);
+
+        _stakedTokenAmount[didAddress] = _stakedTokenAmount[didAddress] + tokenAmount;
+
+        emit TokenDeposited(didAddress, tx.origin, tokenAmount);
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function getNodeIssueFee() external view returns(uint){
+        return _slotInfo.NODE_ISSUE_FEE;
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function updateNodeIssueFee(uint value) external payable onlyOwner {
+        if (value == 0 || value == _slotInfo.NODE_ISSUE_FEE) {
+            revert InvalidValue();
+        }
+        uint orgFee = _slotInfo.NODE_ISSUE_FEE;
+        _slotInfo.NODE_ISSUE_FEE = value;
+
+        emit UpdateNodeIssueFee(orgFee, value);
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function getTotalIssueFee() external view returns(uint) {
+        return _slotInfo.totalIssueFee;
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function withdrawIssueFee(address to, uint amount) external payable onlyOwner {
+        if (amount > _slotInfo.totalIssueFee) {
+            revert InvalidValue();
+        }
+        IERC20Upgradeable(vdaTokenAddress).transfer(to, amount);
+
+        _slotInfo.totalIssueFee = _slotInfo.totalIssueFee - amount;
+
+        emit WithdrawIssueFee(to, amount);
+    }
+
+    function getSameNodeLogDuration() external view returns(uint) {
+        return _slotInfo.SAME_NODE_LOG_DURATION;
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function updateSameNodeLogDuration(uint value) external payable onlyOwner {
+        if (value == 0 || value == _slotInfo.SAME_NODE_LOG_DURATION) {
+            revert InvalidValue();
+        }
+        uint orgVal = _slotInfo.SAME_NODE_LOG_DURATION;
+        _slotInfo.SAME_NODE_LOG_DURATION = value;
+
+        emit UpdateSameNodeLogDuration(orgVal, value);
+    }
+
+    /**
+     * @dev see { IStorageNodeRegistry }
+     */
+    function logNodeIssue(
+        address didAddress,
+        address nodeAddress,
+        uint reasonCode,
+        bytes calldata requestSignature,
+        bytes calldata requestProof
+    ) external {
+        {
+            // Check whether nodeDID is registered
+            uint nodeId = _didNodeId[nodeAddress];
+            if (nodeId == 0) {
+                revert InvalidDIDAddress();
+            }
+
+            bytes memory params = abi.encodePacked(didAddress, nodeAddress, reasonCode);
+            verifyRequest(didAddress, params, requestSignature, requestProof);
+        }
+
+        
+        DIDLogInformation storage logs = _didLogs[didAddress];
+        // Check whether there are 4 logs in 24 horus
```

Comment:
This should be a variable

Reply:
You mean, the number of lgos in 24 hours can be variable?

Reply:
Added `LOG_LIMIT_PER_DAY` Variable in `SlotInfo` struct.